src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/StepHandler.kt

```diff
@@ -0,0 +1,14 @@
+package org.danilopianini.plagiarismdetector.analyzer
+
+/**
+ * An interface modeling a step handler in the analyzer pipeline.
+ * @param I the type of the input.
+ * @param O the type of the output.
+ */
+fun interface StepHandler<I, O> {
```

Comment:
```suggestion
fun interface StepHandler<in I, out O> : (I) -> O {
```

this looks like it can extend a function.
It actually looks exactly like a function, and since there is no boundary at all on the types, it provides no information but "it's a function from I to O".

maybe it is the case to use just a `typealias StepHandler<I, O> = (I) -> O`?